### ISSUE TITLE
`eth_call` inconsistency between geth/reth

### ISSUE THREAD

**Issue by geomad** [paradigmxyz/reth#6701]
### Describe the bug

The following `eth_call` will produce different results between geth and reth:

```
curl -H 'Content-Type: application/json' http://127.0.01:8545 -d '{"method":"eth_call","params":[{"to": "0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696","data":"0xbce38bd70000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000020000000000000000000000000b27308f9f90d607463bb33ea1bebb41c27ce5ab6000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000a4cdca1753000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000003635c9adc5dea00000000000000000000000000000000000000000000000000000000000000000002bc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2000bb80e236c3932887cdc57c887dd97dec82236d1b40000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"}, "0x1260f3c"],"id":1,"jsonrpc":"2.0"}'
```

geth result is:
```
0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000003bba38c5243e639339
```

while reth:
```
"0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000006408c379a000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000002544600000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
```

the call will be identical to geth, if `"gas" : "0x5f5e100"` is also provided to the call, so my guess is reth is enforcing a gas limit, while geth does not.

### Steps to reproduce

eth_call

### Platform(s)

Linux (x86)

### What version/commit are you on?

reth Version: 0.1.0-alpha.19 Commit SHA: 99d1d8daf

### What database version are you on?

1

### What type of node are you running?

Archive (default)

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by mattsse** [comment]
I believe the gas cap is different:

https://github.com/ethereum/go-ethereum/blob/fc380f52ef9778e988266f776b9593ce719cf79d/eth/ethconfig/config.go#L66-L66

no clue why this is the default though

**Comment by geomad** [comment]
> I believe the gas cap is different:
> 
> https://github.com/ethereum/go-ethereum/blob/fc380f52ef9778e988266f776b9593ce719cf79d/eth/ethconfig/config.go#L66-L66
> 
> no clue why this is the default though

I think the call above failed up to 80-90m gas, so it should fail with geth as well if 50m gas was enforced?

**Comment by geomad** [comment]
Bump

**Comment by mattsse** [comment]
could you please provide more context for the return value?
is this a revert?

**Comment by geomad** [comment]
> is this a revert?

This is a `tryAggregate` call to the `multicall2 (0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696)` contract, trying to quote a uniswap 3 price through the uniswap Quoter contract using the `quoteExactInput` method.
While the successful call (geth without defining gas limit or reth with 100m gas limit) returns 
```
[
  [
    true,
    '0x00000000000000000000000000000000000000000000003bba38c5243e639339'
  ]
]
```
indicating with `true` that the call to the Quoter was successful, the failed call (reth without defining gas limit) returns
```
[
  [
    false,
    '0x08c379a0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000025446000000000000000000000000000000000000000000000000000000000000'
  ]
]
```
which indicates that the call from multicall to the Quoter reverted, with the error message translating to `TF` which is the explained by uniswap docs as : `Transfer Failed : errors with TF if transfer fails`

The Quoter  contract simulates a transaction to get the actual price, and the specific transaction uses a lot of gas, so my guess is that the transfer in the simulation failed because it run out of gas.

TLDR: call does not revert itself, but internal call reverts because it runs out of gas.

**Comment by mattsse** [comment]
I assume this uses gasLeft then?

could you check if your geth instance uses the default gas cap

https://github.com/ethereum/go-ethereum/blob/ba2dd9385c2a51134e520083dc732787a813b107/eth/ethconfig/config.go#L66-L66

if you scale up the default gas cap via cli arg, does it succeed?

https://github.com/paradigmxyz/reth/blob/30d79a98aabcc6385beb6548f7cce9ebd85347ca/crates/node-core/src/args/rpc_server_args.rs#L162-L170

**Comment by geomad** [comment]
> I assume this uses gasLeft then?
> 
> could you check if your geth instance uses the default gas cap
> 
> https://github.com/ethereum/go-ethereum/blob/ba2dd9385c2a51134e520083dc732787a813b107/eth/ethconfig/config.go#L66-L66
> 
> if you scale up the default gas cap via cli arg, does it succeed?
> 
> https://github.com/paradigmxyz/reth/blob/30d79a98aabcc6385beb6548f7cce9ebd85347ca/crates/node-core/src/args/rpc_server_args.rs#L162-L170

wow, you are right. Just checked and my geth instance gas cap is set to 150M. I had totally forgotten that. This explains everything, sorry about that.

**Comment by onbjerg** [comment]
Confirming here that it was a configuration issue and that we can close this @geomad?
